pinecone/client_test.go

```diff
@@ -317,200 +108,65 @@ func (ts *IntegrationTests) TestDescribeNonExistentIndex() {
 	require.Equal(ts.T(), reflect.TypeOf(err), reflect.TypeOf(&PineconeError{}), "Expected error to be of type PineconeError")
 }
 
-func (ts *IntegrationTests) TestDescribeServerlessIndexSourceTag() {
-	if ts.indexType == "pods" {
-		ts.T().Skip("No serverless index to test")
-	}
-	index, err := ts.clientSourceTag.DescribeIndex(context.Background(), ts.idxName)
-	require.NoError(ts.T(), err)
-	require.Equal(ts.T(), ts.idxName, index.Name, "Index name does not match")
-}
-
-func (ts *IntegrationTests) TestDescribePodIndex() {
-	if ts.indexType == "serverless" {
-		ts.T().Skip("No pod index to test")
-	}
-	index, err := ts.client.DescribeIndex(context.Background(), ts.idxName)
-	require.NoError(ts.T(), err)
-	require.Equal(ts.T(), ts.idxName, index.Name, "Index name does not match")
-}
-
-func (ts *IntegrationTests) TestDescribePodIndexSourceTag() {
-	if ts.indexType == "serverless" {
-		ts.T().Skip("No pod index to test")
-	}
-	index, err := ts.clientSourceTag.DescribeIndex(context.Background(), ts.idxName)
-	require.NoError(ts.T(), err)
-	require.Equal(ts.T(), ts.idxName, index.Name, "Index name does not match")
-}
-
 func (ts *IntegrationTests) TestListCollections() {
 	if ts.indexType == "serverless" {
 		ts.T().Skip("No pod index to test")
 	}
 	ctx := context.Background()
 
-	var collectionNames []string
-	for i := 0; i < 3; i++ {
-		collectionName := uuid.New().String()
-		collectionNames = append(collectionNames, collectionName)
-	}
-
-	defer func(ts *IntegrationTests, collectionNames []string) {
-		for _, name := range collectionNames {
-			err := ts.client.DeleteCollection(ctx, name)
-			require.NoError(ts.T(), err, "Error deleting collection")
-		}
-	}(ts, collectionNames)
-
-	for _, name := range collectionNames {
-		_, err := ts.client.CreateCollection(ctx, &CreateCollectionRequest{
-			Name:   name,
-			Source: ts.idxName,
-		})
-		require.NoError(ts.T(), err, "Error creating collection")
-	}
```

Comment:
This seemed a little brittle, and we don't necessarily need to create multiple collections just to test the list. We can reuse the `ts.collectionName` for checking list functionality.

Reply:
Amazing,  nice thinking!